src/Tasks/Microsoft.NET.Build.Tasks/CacheArtifactParser.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Xml;
+using System.Globalization;
+using System.Xml.Linq;
+using NuGet.Versioning;
+using NuGet.Packaging.Core;
+
+#if PRODUCT
```

Comment:
This logic would need to be duplicated in test sources, there is no other way out.
I believe this is the lesser of the evils